lib/Value/Matrix.pm

```diff
@@ -720,26 +929,43 @@ sub Zero {
 	return $self->make($context, @M);
 }
 
-#
-#  Extract a given row from the matrix
-#
+=head3 C<row>
+
+Extract a given row from the matrix.
+
+Usage:
+
+    my $A1 = Matrix([ [ 1, 2, 3, 4 ], [ 5, 6, 7, 8 ], [ 9, 10, 11, 12 ] ]);
+    $A1->row(2);  # returns the row Matrix [5,6,7,8]
+
+=cut
+
 sub row {
 	my $self = (ref($_[0]) ? $_[0] : shift);
 	my $M    = $self->promote(shift);
 	my $i    = shift;
-	return if $i == 0;
-	$i--   if $i > 0;
+	Value::Error("Row must be a positive integer") unless $i =~ m/^[1-9]\d*$/;
+	$i-- if $i > 0;
 	if ($M->isRow) { return if $i != 0 && $i != -1; return $M }
 	return $M->data->[$i];
 }
 
-#
-#  Extract a given column from the matrix
-#
+=head3 C<column>
+
+Extract a given column from the matrix.
+
+Usage:
+
+    $A1 = Matrix([ [ 1, 2, 3, 4 ], [ 5, 6, 7, 8 ], [ 9, 10, 11, 12 ] ]);
+    $A1->column(2);  # returns the column Matrix [[2],[6],[10]]
+
+=cut
+
 sub column {
```

Comment:
Similar to my comment about `row`, the `column` method could take multiple whole numbers as arguments.

Also I am seeing what I think is undesirable behavior here with 1D matrices. For an nD matrix with n > 1, `$matrix->column(1)` is still an nD matrix. Good. But with a 1D matrix, `$matrix->column(1)` is a `Real` rather than a 1D matrix. I think for consistency it should still be a 1D matrix.